### PR TITLE
tests: Add func.call tests

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: tests/dialects/test_func.py [hunk 1]
```diff
@@ -91,6 +93,45 @@ def test_callable_constructor():
...
+    b = Constant.from_int_and_width(2, i32)
+    # Operation to add these constants
+    c = Addi.get(a, b)
+    ret0 = Return.get(c)
+
+    # Create a region to include a, b, c and return
+    region = Region.from_operation_list([a, b, c, ret0])
+
+    # Use this region to create a func0
+    func0 = FuncOp.from_region("func0", [a.results[0].typ, b.results[0].typ],
```
Comment (math-fehr) [inline tests/dialects/test_func.py:109]: You can replace `a.results[0]` with `a.result`. And similarly with `b`.

File: tests/dialects/test_func.py [hunk 2]
```diff
@@ -91,6 +93,45 @@ def test_callable_constructor():
...
+                               [c.results[0].typ], region)
```
Comment (math-fehr) [inline tests/dialects/test_func.py:110]: Same here.

File: tests/dialects/test_func.py [hunk 3]
```diff
@@ -91,6 +93,45 @@ def test_callable_constructor():
...
+
+    # Use these region to create a func0
+    call0 = Call.get(func0.attributes['sym_name'].data, [a, b],
```
Comment (math-fehr) [inline tests/dialects/test_func.py:110]:
```suggestion
    call0 = Call.get(func0.sym_name.data, [a, b],
```

File: tests/dialects/test_func.py [hunk 4]
```diff
@@ -91,6 +93,45 @@ def test_callable_constructor():
...
+                     [ret0.operands[0].typ])
```
Comment (math-fehr) [inline tests/dialects/test_func.py:111]:
```suggestion
                     [ret0.arguments[0].typ])
```
This doesn't change much, but it makes it more homogeneous with the rest of the code.

File: xdsl/dialects/func.py [hunk 5]
```diff
@@ -67,10 +67,11 @@ class Call(Operation):
     # TODO how do we verify that the types are correct?
 
     @staticmethod
-    def get(callee: Union[str, FlatSymbolRefAttr],
-            operands: List[Union[SSAValue, Operation]],
+    def get(callee: Union[str, FlatSymbolRefAttr], ops: Union[SSAValue,
+                                                              Operation],
             return_types: List[Attribute]) -> Call:
-        return Call.build(operands=[operands],
+        ops = [op for op in ops] if ops != () else []
```
[the remark this answers is not in the document]
Comment (webmiche) [reply xdsl/dialects/func.py:73]: I am not quite sure what we should do with these. I guess the optimal way would be if you could fix them, but I fear that you might even have introduced them in the first place and then you have to fix problems of someone else...
What do others think?

File: xdsl/dialects/func.py [hunk 6]
```diff
@@ -67,10 +67,11 @@ class Call(Operation):
     # TODO how do we verify that the types are correct?
 
     @staticmethod
-    def get(callee: Union[str, FlatSymbolRefAttr],
-            operands: List[Union[SSAValue, Operation]],
+    def get(callee: Union[str, FlatSymbolRefAttr], ops: List[Union[SSAValue,
+                                                                   Operation]],
             return_types: List[Attribute]) -> Call:
-        return Call.build(operands=[operands],
+        ops = [op for op in ops] if ops != () else []
```
Comment (superlopuh) [inline xdsl/dialects/func.py:73]: I think a part of the confusion of the type checker is reusing the ops name. In either case, `ops_list = list(ops)` also would work

File: xdsl/dialects/func.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (superlopuh) [reply xdsl/dialects/func.py:73]: Seems worth fixing. @georgebisbas, do you see these pyright issues locally?

**Review by superlopuh — CHANGES_REQUESTED** [review]
Could you fix the type issues please?

File: xdsl/dialects/func.py [hunk 7]
```diff
@@ -67,10 +67,11 @@ class Call(Operation):
     # TODO how do we verify that the types are correct?
 
     @staticmethod
-    def get(callee: Union[str, FlatSymbolRefAttr],
-            operands: List[Union[SSAValue, Operation]],
+    def get(callee: Union[str, FlatSymbolRefAttr], ops: List[Union[SSAValue,
+                                                                   Operation]],
             return_types: List[Attribute]) -> Call:
-        return Call.build(operands=[operands],
+        ops = list(ops)
```
Comment (superlopuh) [inline xdsl/dialects/func.py:73]: do you need this? Is it just to copy the list?

**Review by superlopuh — APPROVED** [review]

**Review by webmiche — APPROVED** [review]

File: xdsl/dialects/func.py (hunk 7 above)
Comment (georgebisbas) [reply xdsl/dialects/func.py]: right, thanks!

**Pull request merged**